crates/ide/src/references.rs

```diff
@@ -259,9 +214,12 @@ fn main() {
 }
 "#,
             expect![[r#"
-                Foo Struct FileId(0) 0..26 7..10 Other
+                Foo Struct FileId(0) 0..26 7..10 Name
 
-                FileId(0) 101..104 StructLiteral
+                FileId(0) 32..35 NameRef
+                FileId(0) 88..91 NameRef
+                FileId(0) 101..104 NameRef
+                FileId(0) 109..112 NameRef
             "#]],
```

Comment:
These `StructLiteral` and `EnumLiteral` tests all have changed output now since there is no filtering being done anymore, but if I am not mistaken this was wrong in the first place(maybe only relevant so that rename worked properly before, Im not sure)?

Reply:
Ah no, this is very intentional and something we want to preserve (and maybe even make easier).

In this example the cursor is on the `{`, not on the name of the struct: `struct Foo $0{`. So this should look *only* for places where `Foo` is constructed. This is very useful, as things are constructed not as frequent as they are used, but construction sites are often the most interesting. Admittedly, this feature is half-broken (https://github.com/rust-analyzer/rust-analyzer/issues/2549) and poorly advertised.

Reply:
Oh now I see, I guess this is something I can try to tackle after I'm done with this 😄